cli/formatter/format.go

```diff
@@ -0,0 +1,788 @@
+		return append(slice, value)
+	}
+	slice = append(slice[:index+1], slice[index:]...)
+	slice[index] = value
+
+	return slice
+}
+
+// constructMarkdownHeaderSeparatorRow constructs separator row in markdown notation.
+func constructMarkdownHeaderSeparatorRow(totalRowAmount int) string {
+	var result = "|-"
+	for i := 1; i < totalRowAmount; i++ {
+		result = result + "|-"
+	}
+	result = result + "|"
+
+	return result
+}
+
+// constructMarkdownEmptyRow constructs empty row in markdown notation.
+func constructMarkdownEmptyRow(totalRowAmount int) string {
+	var result = "| "
+	for i := 1; i < totalRowAmount; i++ {
+		result = result + "| "
+	}
+	result = result + "|"
+
+	return result
+}
+
+// renderSingleColumnTable returns table with single column as string.
+func renderSingleColumnTable(renderBatch []renderNode, opts *Opts) string {
+	t := table.NewWriter()
+	var rows []table.Row
+	rows = append(rows, table.Row{"col1"})
+	for _, node := range renderBatch {
+		switch node.(type) {
+		case float64, float32:
+			rows = append(rows, []interface{}{strconv.FormatFloat(node.(float64), 'f', -1, 64)})
+		case nil:
+			rows = append(rows, []interface{}{"nil"})
+		default:
+			rows = append(rows, []interface{}{fmt.Sprintf("%v", node)})
+		}
+	}
+
+	var rowsAmount = len(rows)
+	if opts.TransposeMode {
+		rows = transposeRows(rows)
+	}
+	t.AppendRows(rows)
+	t.Style().Options.SeparateRows = true
+
+	if opts.NoGraphics {
+		t.SetStyle(table.Style{Box: StyleWithoutGraphics})
+	}
+	if opts.ColWidthMax > 0 {
+		colWidthTransformer := text.Transformer(func(val interface{}) string {
+			str := fmt.Sprintf("%v", val)
+			widthMax := opts.ColWidthMax
+			if utf8.RuneCountInString(str) > widthMax {
+				firstLine := string([]rune(str)[:widthMax])
+				remainingLines := string([]rune(str)[widthMax:])
+				firstLine = firstLine + "+"
+				remainingLines = text.InsertEveryN(remainingLines, '+', widthMax-1)
+				return firstLine + remainingLines
+			}
+			return fmt.Sprintf("%v", val)
+		})
+		var columnConfigs []table.ColumnConfig
+		for i := 1; i <= rowsAmount; i++ {
+			columnConfigs = append(columnConfigs,
+				table.ColumnConfig{
+					Number:      i,
+					Transformer: colWidthTransformer,
+					WidthMax:    opts.ColWidthMax,
+				},
+			)
+		}
+		t.SetColumnConfigs(columnConfigs)
+	}
+
+	if opts.TableDialect == MarkdownTableDialect {
+		markdownTable := strings.Split(t.RenderMarkdown(), "\n")
+		if !opts.TransposeMode {
+			markdownTable = insertStrToStrSlice(markdownTable, 0,
+				constructMarkdownEmptyRow(1))
+			markdownTable = insertStrToStrSlice(markdownTable, 1,
+				constructMarkdownHeaderSeparatorRow(1))
+		} else {
+			markdownTable = insertStrToStrSlice(markdownTable, 0,
+				constructMarkdownEmptyRow(rowsAmount))
+			markdownTable = insertStrToStrSlice(markdownTable, 1,
+				constructMarkdownHeaderSeparatorRow(rowsAmount))
+		}
+		var res string
+		for _, v := range markdownTable {
+			res = res + v + "\n"
+		}
+		return res + "\n"
```

Comment:
It seems that the code used in 3 places, please, extract it into a separate function.

Reply:
Done